jax/experimental/jax2tf/jax2tf.py

```diff
@@ -675,6 +675,116 @@ def _conv_general_precision_config_proto(precision):
+    # rhs_spec is represented as a tuple containing the following:
+    # [out_channels, in_channels] + spatial_filter_shape.
+    supported_rhs_shape = ([nb_spatial_dimensions + 1, nb_spatial_dimensions] +
+                           list(range(nb_spatial_dimensions)))
+    if list(rhs_spec) != supported_rhs_shape:
+      return failure("Input filter (RHS) shape format not supported in "
+                     "TensorFlow")
+    # TF only supports same LHS and output data format
+    if lhs_spec != out_spec:
+      return failure("TensorFlow requires the same data format for LHS and "
+                     "output.")
+    # Alphabet extracted from the documentation of tf.conv{1,2,3}d
+    spatial_dim_alphabet = 'DHW'[-nb_spatial_dimensions:]
+    # TF only supports the following data formats:
+    # - [batch_size, in_channels] + input_spatial_shape
+
+    # TODO(bchetioui): TF currently does not support the above on CPU. To avoid
+    # failing on this platform, this path is commented out for now.
+    #if list(lhs_spec) == list(range(len(lhs_spec))):
+    #  return "NC" + spatial_dim_alphabet
+
+    # - [batch_size] + input_spatial_shape + [in_channels]
+    if list(lhs_spec) == ([0, len(lhs_spec) - 1] +
+                          list(range(1, len(lhs_spec) - 1))):
+      return success("N" + spatial_dim_alphabet + "C")
+    return failure("Data format is unsupported by TensorFlow")
+
+  def convert_dilation_and_compute_result(tf_padding, tf_dim_nums):
+    no_dilation = [1] * nb_spatial_dimensions
+      # TODO(bchetioui): is there a generic way to do a transposed atrous
+      # convolution in TensorFlow?
+    if not (list(lhs_dilation) == no_dilation or
+            list(rhs_dilation) == no_dilation):
+      return "Both LHS and RHS dilations are set"
+    # This is a non-dilated or atrous convolution
+    if list(lhs_dilation) == no_dilation:
+      return tf.nn.convolution(
+          lhs, rhs, strides=window_strides, padding=tf_padding,
+          data_format=tf_dim_nums, dilations=rhs_dilation)
+    # TODO(bchetioui): the below path is unreachable for now, as passing a lhs
+    # dilation to this function will result in convert_padding returning None
+    # systematically. This must be investigated further.
+    # Dilation of the LHS is transposed convolution
+    return tf.nn.conv_transpose(
+        lhs, rhs, out_shape, window_strides, padding=tf_padding,
+        data_format=tf_dim_nums, dilations=lhs_dilation)
+
+  tf_padding, error = convert_padding()
+  if tf_padding is None:
+    return error
+  tf_dim_nums, error = convert_dim_nums()
+  if tf_dim_nums is None:
+    return error
+  return convert_dilation_and_compute_result(tf_padding, tf_dim_nums)
+
+# TODO(bchetioui): enabling this flag permits using a conversion path purely
+# based on TF (and not XLA) for _conv_general_dilated in cases when it is
+# possible. It is disabled by default due to a so far unknown bug.
```

Comment:
I think it is worth explaining more here, even if we do not understand what is going on. Otherwise, we ourselves in a few months won't know what we meant. 

One of the tests (actually, what is special about this test?), fails with an assertion if tf.nn.convolution is used. 
The result of tf.nn.convolution has a different shape than what JAX expects, but only when experimental_compile=True. We cannot reproduce it as a pure TF test.